### PR TITLE
[KYUUBI #1880] Fixed: forcedMaxOutputRows extension for join

### LINKED ISSUE #1880: [Bug] The forcedMaxOutputRows extension does not work for statements with join
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

The forcedMaxOutputRows extension does not work for statements with join 

SQL:
```sql
 select t1.*,t2.* from tmp_test t1 right join tmp_orc t2 on t1.a=t2.a;
```

Optimized Logical Plan 

```sql
Join RightOuter, (a#97 = a#99)
:- Filter isnotnull(a#97)
:  +- HiveTableRelation [`tmp_db`.`tmp_test`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#97, b#98], Partition Cols: []]
+- Relation tmp_db.tmp_orc[a#99,b#100] orc
```

Limit rules does not work

### Affects Version(s)

master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by lsm1**
### _Why are the changes needed?_

Fixed #1880 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

image

image

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
hi @lsm1 sorry for the code format issue, can you run `./dev/reformat` ?

**Review by cfmcgrady — APPROVED** [review]

File: dev/kyuubi-extension-spark-common/src/main/scala/org/apache/kyuubi/sql/watchdog/ForcedMaxOutputRowsBase.scala [hunk 1]
```diff
@@ -70,14 +65,14 @@ trait ForcedMaxOutputRowsBase extends Rule[LogicalPlan] {
...
     case _: MultiInstanceRelation => true
+    case _: Join => true
     case _ => false
   }
 
   protected def canInsertLimit(p: LogicalPlan, maxOutputRowsOpt: Option[Int]): Boolean = {
     maxOutputRowsOpt match {
       case Some(forcedMaxOutputRows) => canInsertLimitInner(p) &&
-          !p.maxRows.exists(_ <= forcedMaxOutputRows) &&
-          !isView
```
Comment (ulysses-you) [inline dev/kyuubi-extension-spark-common/src/main/scala/org/apache/kyuubi/sql/watchdog/ForcedMaxOutputRowsBase.scala:80]: why we need this change ?

Comment (lsm1) [reply dev/kyuubi-extension-spark-common/src/main/scala/org/apache/kyuubi/sql/watchdog/ForcedMaxOutputRowsBase.scala:80]: SQL with view will first enter the resolveViews during analyze.The Analyzer's Execute method will be called again, and the analysis of SQL statements with view will recursively call Analyzer's Batches. https://github.com/apache/incubator-kyuubi/pull/1247 add "isView" fixed this.
image

But now Place the ForcedMaxOutputRowsRule in the projectOptimizerRule stage,we do not need this.

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**